controllers/openshift_controller.go

```diff
@@ -595,212 +390,162 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigDeleteRequest() (ctrl.R
-					if contains(r.kataConfig.Status.UnInstallationStatus.Completed.CompletedNodesList, nodeName) {
-						continue
-					}
-
-					if _, ok := r.kataConfig.Spec.KataConfigPoolSelector.MatchLabels["node-role.kubernetes.io/"+machinePool]; !ok {
-						r.Log.Info("Removing the kata pool selector label from the node", "node name ", nodeName)
-						node, err := r.clientset.CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
-						if err != nil {
-							return ctrl.Result{}, err
-						}
-
-						nodeLabels := node.GetLabels()
-
-						for k := range r.kataConfig.Spec.KataConfigPoolSelector.MatchLabels {
-							delete(nodeLabels, k)
-						}
-
-						node.SetLabels(nodeLabels)
-						_, err = r.clientset.CoreV1().Nodes().Update(context.Background(), node, metav1.UpdateOptions{})
-
-						if err != nil {
-							return ctrl.Result{}, err
-						}
-					}
-				}
+		if !isMcDeleted {
+			err = r.Client.Delete(context.TODO(), mc)
+			if err != nil {
+				// error during removing mc, don't block the uninstall. Just log the error and move on.
+				r.Log.Info("Error found deleting machine config. If the machine config exists after installation it can be safely deleted manually.",
+					"mc", mc.Name, "error", err)
 			}
+			// Sleep for MCP to reflect the changes
+			r.Log.Info("Pausing for a minute to make sure worker mcp has started syncing up")
+			time.Sleep(60 * time.Second)
 		}
 
-		r.Log.Info("Making sure parent MCP is synced properly, KataNodeRole=" + machinePool)
-		if _, ok := r.kataConfig.Spec.KataConfigPoolSelector.MatchLabels["node-role.kubernetes.io/"+machinePool]; ok {
-			mc, err := r.newMCForCR(machinePool)
-			var isMcDeleted bool
+		workreMcp := &mcfgv1.MachineConfigPool{}
+		err = r.Client.Get(context.TODO(), types.NamespacedName{Name: machinePool}, workreMcp)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		r.Log.Info("Monitoring worker mcp", "worker mcp name", workreMcp.Name, "ready machines", workreMcp.Status.ReadyMachineCount,
+			"total machines", workreMcp.Status.MachineCount)
+		r.kataConfig.Status.UnInstallationStatus.InProgress.IsInProgress = corev1.ConditionTrue
+		r.clearUninstallStatus()
+		_, result, err2, done := r.updateStatus(machinePool)
+		if !done {
+			r.Log.Info("done returned from updateStatus")
+			return result, err2
+		}
 
-			err = r.Client.Get(context.TODO(), types.NamespacedName{Name: mc.Name}, mc)
-			if err != nil && errors.IsNotFound(err) {
-				isMcDeleted = true
-			} else if err != nil {
-				return ctrl.Result{}, err
-			}
+		if workreMcp.Status.ReadyMachineCount != workreMcp.Status.MachineCount {
+			return ctrl.Result{Requeue: true, RequeueAfter: 15 * time.Second}, nil
+		}
+	} else {
+		if len(r.kataConfig.Status.UnInstallationStatus.InProgress.BinariesUnInstalledNodesList) > 0 {
+			r.Log.Info("Pausing for a minute to make sure parent mcp has started syncing up")
+			time.Sleep(60 * time.Second)
```

Comment:
Yet another sleep that worries me a little bit.

Reply:
yep, same as above. I think this can be done in a more k8s native way, but would rather not change it in this PR.

Reply:
same answer as above. :-)